pkg/controller/elasticsearch/observer/manager.go

```diff
@@ -118,3 +132,20 @@ func (m *Manager) notifyListeners(cluster types.NamespacedName, previousState St
 	// wait for all listeners to be done
 	wg.Wait()
 }
+
+// extractObserverSettings extracts cluster-specific observer settings defined using annotations.
+func (m *Manager) extractObserverSettings(cluster *esv1.Elasticsearch) Settings {
+	settings := Settings{
+		ObservationInterval: annotation.ExtractTimeout(cluster, ObserverIntervalAnnotation, defaultObservationInterval),
+		RequestTimeout:      annotation.ExtractTimeout(cluster, ObserverRequestTimeoutAnnotation, defaultRequestTimeout),
+		Tracer:              m.tracer,
+	}
+
+	clientTimeout := client.Timeout(cluster)
+	if settings.RequestTimeout > clientTimeout {
+		log.Info("Ignoring observer request timeout annotation because it is larger than the client request timeout", "namespace", cluster.Namespace, "es_name", cluster.Name)
+		settings.RequestTimeout = clientTimeout
+	}
```

Comment:
I would not expect this. If I set a specific observer timeout I want it to be used whatever the reconciliation es client timeout is?

Reply:
Observers get an already constructed Elasticsearch client passed to them -- which makes it impossible to change the timeout in the Observer.  Of course, we can create two different constructor functions for the Elasticsearch client to get over that problem. But, I didn't think that was necessary because what is a legitimate reason for having an Observer request timeout that is longer than the standard Elasticsearch client timeout? They are fetching the health information from the cluster and if that doesn't return in a sensible amount of time, then the chances of other API calls succeeding in a shorter time are pretty low as well.

Reply:
Ah - looking at the code I understand technically that's because we use a timeout in the http client, but also an additional context for the observer http calls? 
Having timeouts done one way in the reconciliation (http client timeout) vs. another way in the observer (through a context) has a bit of a smell to me.
Also while I'm not opposed to the argument that you generally want the observer timeout to be shorter than the reconciliation es timeout, it feels a bit like we're imposing a constraint due to the way the code is organized internally.
How about we refactor slightly the observer code to set the timeout on the http client instead of using a context? If needed I think we could add an additional `SetTimeout(timeout time.Duration)` function to the Elasticsearch Client interface so we can still pass the esClient around, and override its timeout at the observer level? But it looks like we could also pass the right timeout at the observer es client creation time?

Reply:
To summarise an offline conversation I had with @sebgl: we both agreed that the special timeout for Observer requests doesn't make much sense and could just be removed.